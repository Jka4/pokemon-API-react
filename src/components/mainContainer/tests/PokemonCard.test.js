import React from "react";
import { render, fireEvent, screen } from '@testing-library/react';
import MainContainer from '@MainContainer';
import { BrowserRouter as Router } from "react-router-dom";
import { getRandomPokemon } from "@APIutils";
import renderer from 'react-test-renderer';

const PokemonCardFetch = () => {

	getRandomPokemon(1);

	return (
		<Router>
			<MainContainer />
		</Router>
	)
}

test('Fetch makes an API call', async () => {
	render(<PokemonCardFetch />)
	const greetingNode = await screen.findByTestId('testId')
	expect(greetingNode).toHaveTextContent('bulbasaur')
})


it('renders correctly', async () => {
	getRandomPokemon(1);

	const tree = renderer
		.create(<PokemonCardFetch />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
