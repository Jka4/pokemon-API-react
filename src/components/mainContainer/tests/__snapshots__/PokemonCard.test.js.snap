// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCardFetch MainContainer renders correctly 1`] = `
<section
  className="main"
>
  <div
    className="getButtonLine"
  >
    <div
      className="Tilt-inner"
    >
      <img
        alt=""
        className="logotype logoBase64"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAWCAYAAACCAs+RAAAGdElEQVRYR+2Wf0xb1xXHz/Pzs9/Dv41/YAM2JmAD4UfITEPCEpJ1tAmRWKOSrGmrVEuibqpapZO6qVGJ5DZao3bZVm1t02hR27XLpJI2EmVVS5sQihNIgSSEHwYbG0zABv+AZ+Nfz88/XuVsKJOiJJSp6tT1/nP1ru45937O99x3DgLfk4F8TzjgB5D/NSX/XxVpRQEsDIApfbsirWgRVLMz63Y4ncjM2QKnnsFY27cZG2uctp8MOZ343xcgyw1gQovgMdQOp1MAphQAMP+twt9AkXFBNmiEC0DEAc6QAHsyF7g5imCCS0KuLMRBJZlvmubMA1jjWVkfbpXJqMfqKurLbGPrO4dn+acIOuYCPCDmMBJ2Oo5FJeDy26E4/i2CmG5GF+ARQs5V5UTi7GIASh0HdkQhYmxzwXkbQEkoAzGHhUuZhMaIcHiGjAVFx6+mwEryeP49ZYVIsz5filuGde1jbnEbgkbCQMfyRUKPBGPhTiqmvuSL/2lq+RwaIZMtT7/u/c1x/r/hypg7Z8Et/DsqIuH17GUjdLLpYX+5vsihSkbnlHyCljin8hI266Ybi0H5F75Rzkcx6YI8leQeNFaN1G82+nIzrs0DcpebVAfQFL9CrwONUhphLg9qx8adxLwMH2I37OgRVJZ6CYulaLTv2o52uz2v3xcXqlSSG9u3bOqW6nM8Poujxm+xaTxKZcxvnVSTc0HxlbupdkcQHtZ7ur7ahvzl1D92FFZ0iJedBGaBGevdmf7s0591dg/++FWPh5ApZOO/Nh07UbP1wY+RWIgAc/tuuDqyORUOCKNqRTiWse0yFxNkNMg7cPBvrMbmD0EoAhg0l0+0vPD6pa5r9YOQiufte+T9hx/6qVkXDPFjo5YN8x0XqpzTc5LL+fKQv89W/sdVgxTnRZhDz/Tv+vkvXsoixDHIQDhtm9OkpxLtMdf5+vqMF2e8YqZm3YXaw8deUResvQpLi2roPrsXLl6qhpLKmahONxYmyRzicnelIBBKwuEXT4J2be/NO7kngT761ElPu3lXwFg+KTz07Ptaw7pBoCgUqIgKhq7ofBc6Kr5wuUtnPrm+/vlVgQjwtv1iXJ9W5ELTUdMfNjQ0nlKMXjcmuztr6ZBnCxaKEWjAm6IyznXagdSDez8jFIYRPOA1wsWzDeCeKYBtD3Ul1HnjtHu2hDM0UIGx4xg07j4HeYavIEiywHE1G0aG74OQ3wAalQe05VMgUc+BRDUNQpEbon7wv/m7Y++0tjdP9TuLT6wKBKBNmTEkOFvqntzXtqvlyNHtKDcmO9++J937ZVOysNjHqd74MeCJCPgXhKBZMwOEiAKXuxDc0/J0oZpiGdZQzEI0mewaFkc8M1LW2lzgy6Uc1tKiDObt2SCUWqGqoR+yJD4g3QIIzMtAIHODuuQCCKUAs7Y8y8En3nlxaKzGseo3skwvgoBEW2Bteu33x5/f1nymZLDzafjrq3vAUGWBA8+dBB5bCNbeLRD2ysF2A5gJb6wfYYe9hoJQvpSRVY3YNXHzDd4/F8Mee062S4kzOXKfbZ2O8WtLG+pHkQd2D4BINwEOhxpcY2rGYLQgZQ1vAY5BpP3MgXPNj7/ZQtNT05k/5CoVuWXGw66t2/3AB6aTJ87t9Nqa2PsfbaTlRR2hXb/8Mo4k9Gn/2AbEPlCAfGXlT7qoyEdRZGEiX+mq5Kd5TTP+8qgrXPhGjO6+hGEDahzL1TDpvB+p+exNG9XREr1hCbEvpBeHHJoUJ52r+NWTZum+Q8cxksy2thz+7Qenz1a/J6cS7nvVmhUWxDalTjby+GvPkgc5idr8I3/OH3dEz58XqYbH0ZQhlo6v4fiXhGycLQrRyYKBIEX5udyzKhRNG4TcijSKKK+7FmtnAUwsAJGQw9GqFPyktpAfVvriNHs2BP4kIyZQhl+2scx6/xNNfVqKqu15r7X53e7JiBmgJHyv6r9CkP0CqRCpq80x7lOIq9deniG7HP6edxOJo9du6Xa39mW5uJqSt6dHBu5fLQ+X+7KeQOcfzeXgRhG6tXciXNfqi3dM/mcXcaf0WiGIiZ0tcBYJucU7udh6zXQg2BuLff4JwNt3zdtVtB0Izn6uHkfxajyrxhYj63uCICZX4ucbgbDS+g0ctDR/KekZDkX7Or8FEMCwI9ViDlHJQ0qnneHSK/d65MuQKwRZSUy+2z0/gHy38b/99K8BURHLNYz4KKoAAAAASUVORK5CYII="
      />
    </div>
  </div>
  <div
    className="Button"
  >
    <a
      aria-current={null}
      href="/allPokemons/"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
           
          SHOW ALL POKEMON
           
        </span>
      </button>
    </a>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained clearBtn MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        CLEAR
      </span>
    </button>
    <span
      className="MuiBadge-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained getPokemons MuiButton-containedSecondary"
        disabled={false}
        id="getPokemons"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Loading...
        </span>
      </button>
      <span
        className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
      >
        1
      </span>
    </span>
  </div>
  <h2
    className="titles"
  >
    <ul
      className="titlesList"
    >
      <li>
        photo
      </li>
      <li>
        name
      </li>
      <li>
        order
      </li>
      <li>
        exp
      </li>
    </ul>
  </h2>
  <div
    className="itemList"
  >
    <div
      className="cardsContainer"
    >
      <div
        className="MuiPaper-root MuiCard-root pokemonCard_Outher MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          aria-current={null}
          data-pokemon_id={1}
          data-testid="testId"
          href="/detailedPage/pokemon/bulbasaur"
          onClick={[Function]}
        >
          <div
            className="pokemonCard"
          >
            <div
              className="imageWrapper"
            >
              <div
                aria-busy="true"
                className=""
                style={Object {}}
              >
                <svg
                  aria-label="audio-loading"
                  height={50}
                  viewBox="0 0 38 38"
                  width={50}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <linearGradient
                      id="a"
                      x1="8.042%"
                      x2="65.682%"
                      y1="0%"
                      y2="23.865%"
                    >
                      <stop
                        offset="0%"
                        stopColor="red"
                        stopOpacity="0"
                      />
                      <stop
                        offset="63.146%"
                        stopColor="red"
                        stopOpacity=".631"
                      />
                      <stop
                        offset="100%"
                        stopColor="red"
                      />
                    </linearGradient>
                  </defs>
                  <g
                    fill="none"
                    fillRule="evenodd"
                  >
                    <g
                      transform="translate(1 1)"
                    >
                      <path
                        d="M36 18c0-9.94-8.06-18-18-18"
                        id="Oval-2"
                        stroke="red"
                        strokeWidth="2"
                      >
                        <animateTransform
                          attributeName="transform"
                          dur="0.9s"
                          from="0 18 18"
                          repeatCount="indefinite"
                          to="360 18 18"
                          type="rotate"
                        />
                      </path>
                      <circle
                        cx="36"
                        cy="18"
                        fill="#fff"
                        r={1}
                      >
                        <animateTransform
                          attributeName="transform"
                          dur="0.9s"
                          from="0 18 18"
                          repeatCount="indefinite"
                          to="360 18 18"
                          type="rotate"
                        />
                      </circle>
                    </g>
                  </g>
                </svg>
              </div>
            </div>
            <span>
              bulbasaur
            </span>
            <span>
              1
            </span>
            <span>
              64
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</section>
`;
